tests/role/test_role.yml

```diff
@@ -0,0 +1,377 @@
+---
+- name: Test role module
+  hosts: ipaserver
+  become: yes
+  gather_facts: yes
+
+  tasks:
+  - name: Set environment facts.
+    import_tasks: env_facts.yml
+
+  - name: Setup environment.
+    import_tasks: env_setup.yml
+
+  # tests
+  - name: Ensure role is present.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: renamerole
+      description: A role in IPA.
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure role is present, again.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: renamerole
+      description: A role in IPA.
+    register: result
+    failed_when: result.changed
+
+  - name: Rename role.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: renamerole
+      rename: testrole
+    register: result
+    failed_when: not result.changed
+
+  - name: Rename role, again.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: renamerole
+      rename: testrole
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure role has member has privileges.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: testrole
+      privilege:
+      - DNS Servers
+      - Host Administrators
+      action: member
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure role has member has privileges, again.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: testrole
+      privilege:
+      - DNS Servers
+      - Host Administrators
+      action: member
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure role has less privileges.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: testrole
+      privilege:
+      - Host Administrators
+      action: member
+      state: absent
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure role has less privileges, again.
+    iparole:
+      ipaadmin_password: SomeADMINpassword
+      name: testrole
+      privilege:
+      - Host Administrators
+      action: member
+      state: absent
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure role has member has privileges restored.
```

Comment:
Missing idempotency check for this test.

Reply:
Fixed.